examples/rosa-hcp-public-with-byo-vpc-byo-iam-byo-oidc/main.tf

```diff
@@ -0,0 +1,53 @@
+locals {
+  account_role_prefix  = "${var.cluster_name}-account"
+  operator_role_prefix = "${var.cluster_name}-operator"
+}
+
+module "hcp" {
+  source = "../../"
+
+  cluster_name         = var.cluster_name
+  openshift_version    = var.openshift_version
+  machine_cidr         = var.machine_cidr
+  account_role_prefix  = module.account_iam_resources.account_role_prefix
+  operator_role_prefix = module.operator_roles.operator_role_prefix
+  oidc_config_id       = module.oidc_config_and_provider.oidc_config_id # replace with variable once split out properly
+  aws_subnet_ids         = concat(module.vpc.public_subnets, module.vpc.private_subnets)
+  aws_availability_zones = module.vpc.availability_zones
+  path                   = module.account_iam_resources.path
+  replicas               = length(module.vpc.availability_zones)
+  depends_on = [ module.vpc ]
```

Comment:
@nirarg Would you suggest a different approach? I couldn't get it to work properly otherwise

Reply:
In order to wait for the creation completion of all resources in vpc sub-module we use the [time_sleep resource](https://github.com/terraform-redhat/terraform-rhcs-rosa-hcp/blob/2fb5724ecfaf8b05ec1ddb2e0b854e74574dc829/modules/vpc/main.tf#L203).
In this case, the dependencies you have don't go through the time_sleep resource (`public_subnets`, `private_subnets` and `availability_zones`).
In order to fix this, you can use one of the options:
1. Change the outputs of `public_subnets`, `private_subnets` and `availability_zones` to be used from the time_sleep resource
2. Change variable assignment of `machine_cidr` to use the output from vpc (`machine_cidr = module.vpc.cidr_block`)

Once you complete one of the options above you can remove the `depends_on`

Reply:
I had an issue with setting the public/private subnets in the time_sleep as it generates a list of strings but the input should be a single string in the end. Not sure how to convert it through the terraform management file

Let me try the machine cidr assignment